File: 1138-grumpy-bookstore-owner/grumpy-bookstore-owner.java
class Solution {
    public int maxSatisfied(int[] customers, int[] grumpy, int minutes) {
        int x = 0;
        for (int i = 0; i < customers.length; i++) {
            if (grumpy[i] == 0) {
                x += customers[i];
            }
        }
        int y = 0;
        int z = 0;
        for (int i = 0; i < customers.length; i++) {
            z += grumpy[i] == 1 ? customers[i] : 0;
            if (i >= minutes) {
                z -= grumpy[i - minutes] == 1 ? customers[i - minutes] : 0;
            }
            y = Math.max(y, z);
        }
        return x + y;
    }
}

File: 1138-grumpy-bookstore-owner/grumpy-bookstore-owner.py
class Solution(object):
    def maxSatisfied(self, customers, grumpy, minutes):
        x = sum([customers[i] for i in range(len(customers)) if grumpy[i] == 0])
        y = 0
        z = 0

        for i in range(len(customers)):
            z += customers[i] if grumpy[i] == 1 else 0
            if i >= minutes:
                z -= customers[i - minutes] if grumpy[i - minutes] == 1 else 0
            y = max(y, z)
        
        return x + y
